lib/imapSimple.js

```diff
@@ -319,6 +320,64 @@ ImapSimple.prototype.addMessageLabel = function (source, labels, callback) {
 };
 
 /**
+ * Adds the provided flag(s) to the specified message(s).
+ *
+ * @param {string|Array} uid The messages uid
+ * @param {string|Array} flags Either a single string or an array of strings indicating the flags to add to the
+ *  message(s).
+ * @param {function} [callback] Optional callback, receiving signature (err)
+ * @returns {undefined|Promise} Returns a promise when no callback is specified, resolving when the action succeeds.
+ * @memberof ImapSimple
+ */
+ImapSimple.prototype.addFlags = function (uid, flags, callback) {
+    var self = this;
+
+    if (callback) {
+        return nodeify(self.addFlags(uid, flags), callback);
+    }
+
+    return new Promise(function (resolve, reject) {
+        self.imap.addFlags(uid, flags, function (err) {
+            if (err) {
+                reject(err);
+                return;
+            }
+
+            resolve();
+        });
+    });
+};
+
+/**
+ * Removes the provided flag(s) to the specified message(s).
+ *
+ * @param {string|Array} uid The messages uid
+ * @param {string|Array} flags Either a single string or an array of strings indicating the flags to remove from the
+ *  message(s).
+ * @param {function} [callback] Optional callback, receiving signature (err)
+ * @returns {undefined|Promise} Returns a promise when no callback is specified, resolving when the action succeeds.
+ * @memberof ImapSimple
+ */
+ImapSimple.prototype.delFlags = function (uid, flags, callback) {
+    var self = this;
+
+    if (callback) {
+        return nodeify(self.delFlags(uid, flags), callback);
+    }
+
+    return new Promise(function (resolve, reject) {
+        self.imap.delFlags(uid, flags, function (err) {
+            if (err) {
+                reject(err);
+                return;
+            }
+
+            resolve();
+        });
+    });
+};
```

Comment:
Added something to the readme. Feel free to change the text if you don't like or can improve it